### DISCUSSION TITLE
"Following Only" filter for activity page notifications

### DISCUSSION THREAD

**Discussion by karleaux** [AprilSylph/XKit-Rewritten#543] [Suggestion Box]
Hello! Is it possible to add a filter for the activity page so that it only shows notifications from people you follow? I remember being able to do this a few years back, but I wasn't sure if that was native Tumblr functionality or a previous XKit thing. No idea why they even removed it.

I've had _some_ success doing this albeit in a very hacky way (using JS to hide notifs without the 'following' blue highlight CSS class) but it does fail when you have the 'Group similar notifications' setting turned on (plus it still keeps the hidden notifs the DOM). I figured there was a better way if you had access to the actual notification object/response?

**Comment by AprilSylph** [comment]
It was a thing in New XKit, and it wasn't removed - it was broken by Tumblr's React overhaul of the activity page.

Grouped notifications do not have any data for if you're following any of the shown blogs. It's also not safe to remove anything from the DOM on a React page from outside the React code.

So, I agree this would be a nice feature, but it would essentially work exactly the same way as what you're describing. Although, the JS part could be cut out for the most part - JS is needed to grab the CSS map, but once that's done, notifications could be hidden with just CSS using [:not()](https://developer.mozilla.org/en-US/docs/Web/CSS/:not).

**Reply by karleaux** [reply]
Oooh I see, thanks! And yeah, the JS was mostly to insert a toggle button so I could test, lol. 

So if CSS is the route, I guess the major obstacle would be automatically shifting to a non-grouped notification feed beforehand, so it properly filters all the notifs with the CSS rule? Would that be possible through, say, sending the same request as the existing activity filter does (if it even does that?) except with the grouping setting turned off? Unless there's a more elegant solution available.

**Reply by AprilSylph** [reply]
That isn't possible, no. The filters on the activity page aren't a bunch of account or blog settings; they're local. These local settings are then used to construct each individual API request to fetch that blog's notifications. It would be irresponsible to attempt to alter these local settings, if it's possible at all.
